### ISSUE TITLE
ZSH Completions do not work

### ISSUE THREAD

**Issue by allemangD** [f3d-app/f3d#1361]
ZSH completions do not function, with error `_arguments:comparguments:327: invalid argument: loading[the first file, use with --verbose ]`

The commands in [completion.zsh](https://github.com/f3d-app/f3d/blob/master/resources/completion.zsh) produce `arguments` list with contents:

```
...
--no-render[Do not render anything and quit right after ]
loading[the first file, use with --verbose ]
--max-size[<size in Mib>  Maximum size in Mib of a file to load, ]
...
--quiet[Enable quiet mode, which supersede any ]
using[`--verbose=quiet` instead)]
--progress[Show progress bar]
```

Where the lines beginning with `loading` and `using` are not valid options to `_arguments`. (And the help-text for multi-line blocks like `--no-render` or `--max-size` or `--quiet` are truncated and not very helpful)

They are included because the help-text for `--no-render` and `--quiet` include `--`, so their help-text contents are mangled into invalid options. See:

https://github.com/f3d-app/f3d/blob/b53ba1ae3a60e244d55291ae1598df81635b8366/application/F3DOptionsParser.cxx#L312

https://github.com/f3d-app/f3d/blob/b53ba1ae3a60e244d55291ae1598df81635b8366/application/F3DOptionsParser.cxx#L321

**To reproduce:**

1. Open a `zsh` shell
2. Type `f3d`
3. Press tab (or other configured completion key)

**System Information:**

Ubuntu 22.04

- zsh 5.9
- f3d 2.3.0

MacOS 12.7

- zsh 5.8
- f3d 2.3.1

**Additional Context:**

In Nov 2022 I created https://github.com/f3d-app/f3d/pull/462 to get zsh completion partially working. At some point I upgraded and the mentioned `--no-render` and `--quiet` changes broke it again. In that PR I attempted to merge multi-line helptext, but failed due to compatibility issues on MacOS. That multi-line merging should fix this issue and also enhance completion helptext, since only the start-of-line `-short` and `--long` options would be matched; `--` in helptext would be only part of the helptext.

So since I have my MacOS environment working to test compatibility locally, I'll spend some time fixing that multi-line support code and create a PR if I'm successful. I'm not sure how much availability I'll have to work on that and when the PR will come, but I wanted to log the issue and describe the bug here for now.

My original attempt at multi-line support is at https://github.com/f3d-app/f3d/commit/bc78f68ae15b87daf510aaafa19cc18dc96f5200; but it does not function on MacOS.

**Comment by mwestphal** [comment]
@Meakk didn't you fix this recently?

**Comment by mwestphal** [comment]
the issue looks pretty similar though, would the fix be similar ?

https://github.com/f3d-app/f3d/pull/1325

**Comment by allemangD** [comment]
I think _a_ fix can probably be done with the same strategy. It looks like the fish completions still truncate multi-line helptext though, it might be nice to implement that and apply that fix to both fish and zsh completions.

However since the fix for #1325 was so small I'll check if a similar change can be applied to zsh for a quick patch.

**Comment by mwestphal** [comment]
I'll merge this since it is fixed but feel free to open another feature issue to improve the auto completion as you suggested @allemangD
